web3.md

```diff
@@ -1086,19 +1089,41 @@ Collecting Coverage Data
       [priority(25)]
 
     rule <k> #initVM ... </k>
-         <opcodeCoverage> OC => OC [ {keccak(CODE) | EPHASE} <- .Set ] </opcodeCoverage>
+         <opcodeCoverage> OC => OC [ {keccak(PGM) | EPHASE} <- .Set ] </opcodeCoverage>
          <execPhase> EPHASE </execPhase>
-         <program> CODE </program>
-      requires notBool {keccak(CODE) | EPHASE} in_keys(OC)
+         <program> PGM </program>
+      requires notBool {keccak(PGM) | EPHASE} in_keys(OC)
       [priority(25)]
 
+
+    rule <k> #initVM ... </k>
+         <opcodeLists> OL => OL [ {keccak(PGM) | EPHASE} <- #getOpcodeList(PGM,SCHED) ] </opcodeLists>
+         <execPhase> EPHASE </execPhase>
+         <schedule> SCHED </schedule>
+         <program> PGM </program>
+      requires notBool {keccak(PGM) | EPHASE} in_keys(OL)
+      [priority(25)]
+
+    syntax OpcodeItem ::= "{" Int "|" OpCode "}"
+
+    syntax List ::= #getOpcodeList( ByteArray, Schedule ) [function]
+ // ----------------------------------------------------------------
+    rule #getOpcodeList(PGM , SCHED) => #parseByteCode(0, #sizeByteArray(PGM), PGM, SCHED)
+
+    syntax List ::= #parseByteCode ( Int, Int, ByteArray, Schedule ) [function]
+ // ---------------------------------------------------------------------------
+    rule #parseByteCode(PCOUNT, SIZE, _, _) => .List
+      requires PCOUNT >=Int SIZE
+    rule #parseByteCode(PCOUNT, SIZE, PGM, SCHED) => ListItem({ PCOUNT | #dasmOpCode(PGM [ PCOUNT ], SCHED) } ) #parseByteCode(PCOUNT +Int #widthOp(#dasmOpCode(PGM [ PCOUNT ], SCHED)), SIZE, PGM, SCHED)
+      requires PCOUNT <Int SIZE
```

Comment:
We should probably make this tail-recursive.

Something like `#parseByteCode(PGM, SCHED) => #parseByteCodeAux(0, #sizeByteArray(PGM), PGM, SCHED, .List)`, and then the rules for `#parseByteCodeAux` build the resulting `List` as the last argument.